sling/pyapi/pyarray.cc

```diff
@@ -116,8 +191,14 @@ int PyArray::Contains(PyObject *key) {
 
   // Check if value is contained in array.
   ArrayDatum *arr = array();
-  for (int i = 0; i < arr->length(); ++i) {
-    if (arr->get(i) == handle) return true;
+  if (slice == nullptr) {
+    for (int i = 0; i < arr->length(); ++i) {
+      if (pystore->store->Equal(arr->get(i), handle)) return true;
+    }
+  } else {
+    for (int idx = slice->start; idx != slice->stop; idx += slice->step) {
+      if (pystore->store->Equal(arr->get(pos(idx)), handle)) return true;
```

Comment:
do you need to use pos() here? The loop is from slice->start to slice->stop and not from 0 to length().